### PR TITLE
Adding non-ephemeral canary

### PR DESCRIPTION
**Pull request by gautamkmr**
This change adds support to run canary on non-ephemeral cluster. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/codebuild/run_canarytest.sh [hunk 1]
```diff
@@ -49,11 +49,20 @@ echo "Launching canary test for ${COMMIT_SHA}"
...
 # Launch EKS cluster if we need to and define cluster_name,CLUSTER_REGION.
 echo "Launching the cluster"
-readonly cluster_name="sagemaker-k8s-pipeline-"$(date '+%Y-%m-%d-%H-%M-%S')""
 
-# By default eksctl picks random AZ, which time to time leads to  capacity issue.
-# Generally 1a, 1b, 1c are topmost available AZ, hence specifying it explicitly 
-eksctl create cluster "${cluster_name}" --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version ${CLUSTER_VERSION} 
+cluster_name="sagemaker-k8s-canary-"$(date '+%Y-%m-%d-%H-%M-%S')""
+
+if [ -z "${NON_EPHEMERAL_CLUSTER}" ]
```
Comment (RedbackThomson) [inline tests/codebuild/run_canarytest.sh]: Could we make this a more generic variable name?
Something like `USE_EXISTING_CLUSTER`?

File: tests/codebuild/run_canarytest.sh [hunk 2]
```diff
@@ -49,11 +49,20 @@ echo "Launching canary test for ${COMMIT_SHA}"
 
 # Launch EKS cluster if we need to and define cluster_name,CLUSTER_REGION.
 echo "Launching the cluster"
-readonly cluster_name="sagemaker-k8s-pipeline-"$(date '+%Y-%m-%d-%H-%M-%S')""
 
-# By default eksctl picks random AZ, which time to time leads to  capacity issue.
-# Generally 1a, 1b, 1c are topmost available AZ, hence specifying it explicitly 
-eksctl create cluster "${cluster_name}" --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version ${CLUSTER_VERSION} 
+cluster_name="sagemaker-k8s-canary-"$(date '+%Y-%m-%d-%H-%M-%S')""
```
Comment (RedbackThomson) [inline tests/codebuild/run_canarytest.sh]: Should this move inside the if statement?

Comment (gautamkmr) [reply tests/codebuild/run_canarytest.sh]: Why? we overwrite the name based on condition so I think its fine.

File: tests/codebuild/run_canarytest.sh (hunk 1 above)
Comment (gautamkmr) [reply tests/codebuild/run_canarytest.sh]: I think this name is good and explanatory enough.

File: tests/codebuild/run_canarytest.sh [hunk 3]
```diff
@@ -49,11 +49,20 @@ echo "Launching canary test for ${COMMIT_SHA}"
...
+if [ -z "${NON_EPHEMERAL_CLUSTER}" ]
+then
+   eksctl_args=( --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version "${CLUSTER_VERSION}" )
+   [ "${CLUSTER_PUBLIC_SUBNETS}" != "" ] && eksctl_args+=( --vpc-public-subnets="${CLUSTER_PUBLIC_SUBNETS}" )
+   [ "${CLUSTER_PRIVATE_SUBNETS}" != "" ] && eksctl_args+=( --vpc-private-subnets="${CLUSTER_PRIVATE_SUBNETS}" )
+
+   eksctl create cluster "${cluster_name}" "${eksctl_args[@]}"
+else
+   cluster_name="non-ephemeral-cluster"
+   aws eks update-kubeconfig --name "${cluster_name}" --region "${CLUSTER_REGION}"
```
Comment (cadedaniel) [inline tests/codebuild/run_canarytest.sh]: Does this place the updated kubeconfig in `~/.kube/eksctl/clusters/<name>` or `~/.kube/config`?

File: tests/codebuild/run_canarytest.sh (hunk 1 above)
Comment (cadedaniel) [reply tests/codebuild/run_canarytest.sh]: ++ to Nick's suggestion; it more accurately describes the purpose as this code uses any existing cluster, not necessarily a non-ephemeral one.

File: tests/codebuild/run_canarytest.sh (hunk 3 above)
Comment (RedbackThomson) [reply tests/codebuild/run_canarytest.sh]: From the awscli docs:
```
Added new context arn:aws:eks:us-west-2:012345678910:cluster/example to /Users/ericn/.kube/config
```

Comment (cadedaniel) [reply tests/codebuild/run_canarytest.sh]: Thanks for looking up for me :) So will this work with line 68?

```
export KUBECONFIG="/root/.kube/eksctl/clusters/${cluster_name}"
```

Comment (RedbackThomson) [reply tests/codebuild/run_canarytest.sh]: Why would we need to export that env var? What uses that?

Comment (gautamkmr) [reply tests/codebuild/run_canarytest.sh]: yeah I am not sure if we need env in second case.

File: tests/codebuild/run_canarytest.sh [hunk 4]
```diff
@@ -5,11 +5,7 @@
 # Build environment `Docker image` has all prerequisite setup and credentials are being passed using AWS system manager
 
 CLUSTER_REGION=${CLUSTER_REGION:-us-east-1}
-CLUSTER_VERSION=${CLUSTER_VERSION:-1.13}
-
-# Define the list of optional subnets for the EKS test cluster
-CLUSTER_PUBLIC_SUBNETS=${CLUSTER_PUBLIC_SUBNETS:-}
-CLUSTER_PRIVATE_SUBNETS=${CLUSTER_PRIVATE_SUBNETS:-}
```
Comment (gautamkmr) [inline tests/codebuild/run_canarytest.sh]: something is messed up here.

**Pull request merged**